src/metabase/notification/send.clj

```diff
@@ -388,14 +390,47 @@
   "The default options for sending a notification."
   {:notification/sync? false})
 
+(setting/defsetting notification-cutoff-timestamp
+  "Timestamp that serves as an anchor point for notifications.
+  Timestamp should be ISO 8601 format and in UTC.
+  Tip: Use the `date -u -Iseconds` command to get the current time in UTC.
+
+  Used for staging instances to skip sending existing notifications."
+  :type       :string
+  :default    nil
+  :encryption :no
+  :export?    false
+  :cache? false
+  :getter     (fn []
+                (when-let [timestamp (some-> (setting/get-value-of-type :string :notification-cutoff-timestamp)
+                                             u.date/parse ;; expects ISO 8601 format
+                                             (#(.toOffsetDateTime ^ZonedDateTime %)))]
+                  (assert (t/< timestamp (t/offset-date-time)) "Cutoff timestamp must be in the past")
+                  timestamp))
+  :visibility :internal)
+
+(defn- skip-notification-because-of-cutoff?
+  "If `notification-cutoff-timestamp2` is set, returns true for notifications that have updated_at older than the cutoff timestamp.
+
+  This should be done on staging instances only.
+  IMPORTANT: DO NOT LET THIS MERGE TO MASTER."
```

Comment:
why not tho, seems generally useful for staging/demo instances?

Reply:
Yeah I would use this when developing against master, e.g. when wanting to benchmark a backfill against staging.